app/src/main/java/polyrooms/polyrooms/MapsActivity.kt

```diff
@@ -35,21 +37,25 @@ class MapsActivity : AppCompatActivity() {
                 // Map is set up and the style has loaded. Now you can add data or make other map adjustments
 
                 val time = intent.extras.get("time") as Time
-                println(time)
-
-                /*
-
-                Call this function rather than just setMarkers once database is up and running
 
                 viewModel.getBuildings(time).observe(this, Observer { buildings ->
                     setMarkers(mapboxMap, buildings)
-                }) */
+                })
 
-                setMarkers(mapboxMap, null)
                 mapboxMap.setOnMarkerClickListener { marker ->
                     Toast.makeText(this, marker.title, Toast.LENGTH_LONG).show()
-                    val intent = Intent(this, MainActivity::class.java)
-                    intent.putExtra("building", marker.title)
+                    val intent = Intent(this, ReserveActivity::class.java)
+
+                    val buildings: List<Building>? = viewModel.getBuildings(time).value
```

Comment:
If you want to optimize this, you could attach each marker to its corresponding Building data object. That way you wouldn't have to query the entire database to get that data again.